### PR TITLE
Analyzer: fix 01019_alter_materialized_view_consistent

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 5703e0995bf6d9b55ccb354a5248ef5e0e7cd686 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56246/5703e0995bf6d9b55ccb354a5248ef5e0e7cd686/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6731299810">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56246/5703e0995bf6d9b55ccb354a5248ef5e0e7cd686/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Storages/StorageMaterializedView.cpp [hunk 1]
```diff
@@ -283,8 +283,6 @@ void StorageMaterializedView::alter(
         const auto & old_select = old_metadata.getSelectQuery();
 
         DatabaseCatalog::instance().updateViewDependency(old_select.select_table_id, table_id, new_select.select_table_id, table_id);
-
-        new_metadata.setSelectQuery(new_select);
```
Comment (vdimir) [inline src/Storages/StorageMaterializedView.cpp:287]: This line just sets the same `select` to `new_metadata`, see line above
```cpp
const auto & new_select = new_metadata.select;
```

File: src/Processors/Transforms/buildPushingToViewsChain.cpp [hunk 2]
```diff
@@ -493,7 +506,8 @@ static QueryPipeline process(Block block, ViewRuntimeData & view, const ViewsDat
         pipeline = interpreter.buildQueryPipeline();
     }
     else
-    {   InterpreterSelectQuery interpreter(view.query, local_context, SelectQueryOptions());
+    {
+        InterpreterSelectQuery interpreter(view.query, local_context, SelectQueryOptions());
```
Comment (vdimir) [inline src/Processors/Transforms/buildPushingToViewsChain.cpp:510]: `InterpreterSelectQuery` checks that view source matches the query inside. But I suppose it's better to have it on `buildPushingToViewsChain` level.

**Review by KochetovNicolai — APPROVED** [review]

**Comment by vdimir** [comment]
> Integration tests (asan, analyzer) [2/6] — fail: 1, passed: 316 [Details](https://s3.amazonaws.com/clickhouse-test-reports/56246/5703e0995bf6d9b55ccb354a5248ef5e0e7cd686/integration_tests__asan__analyzer__[2_6].html)

https://github.com/ClickHouse/ClickHouse/issues/55090 (or it's broken only on analyzer?)

> Stateless tests (release, analyzer) — fail: 1, passed: 5787, skipped: 11 [Details](https://s3.amazonaws.com/clickhouse-test-reports/56246/5703e0995bf6d9b55ccb354a5248ef5e0e7cd686/stateless_tests__release__analyzer_.html)

`01287_max_execution_speed` https://github.com/ClickHouse/ClickHouse/issues/56052

> Upgrade check (debug) — Logical error thrown (see clickhouse-server.log or logical_errors.txt) [Details](https://s3.amazonaws.com/clickhouse-test-reports/56246/5703e0995bf6d9b55ccb354a5248ef5e0e7cd686/upgrade_check__debug_.html)

https://github.com/ClickHouse/ClickHouse/issues/55864

**Pull request merged**